lib/fhir_client/client.rb

```diff
@@ -323,17 +353,29 @@ def parse_reply(klass, format, response)
             else
               FHIR::DSTU2::Json.from_json(response.body)
             end
-          elsif(@fhir_version == :r4 || klass&.ancestors&.include?(FHIR::Model))
+          elsif(@fhir_version == :stu3 || klass&.ancestors&.include?(FHIR::STU3::Model))
```

Comment:
For all of these, ditch `klass&.ancestors&.include?(FHIR::FOO::MODEL)` and just do `klass < FHIR::FOO::Model`.